honeycomb/honeycomb.go

```diff
@@ -423,6 +423,28 @@ func NewExporter(config Config, opts ...ExporterOption) (*Exporter, error) {
 	}, nil
 }
 
+// Consume from the response queue, calling the onError callback when errors
+// are encountered.
+//
+// This method will block until the passed context.Context is cancelled, or
+// until exporter.Close is called.
+func (e *Exporter) RunErrorLogger(ctx context.Context) {
+	responses := libhoney.TxResponses()
+	for {
+		select {
+		case r, ok := <-responses:
+			if !ok {
+				return
+			}
+			if r.Err != nil {
+				e.onError(r.Err)
+			}
+		case <-ctx.Done():
+			break
```

Comment:
Only one nit remains: Here we `break`, whereas on line 437 we `return`. It would be less mysterious to future readers to use the same statement for both cases. Here, I recommend `return`.

Reply:
Good eye. I'll fix that and then call this done, pending approval from @lizthegrey as well.